code/Model/EditableFormField/EditableFileField.php

```diff
@@ -60,14 +64,21 @@ public function getCMSFields()
             )
         );
 
-        $fields->addFieldToTab("Root.Main", LiteralField::create(
-            'FileUploadWarning',
-            '<p class="alert alert-info">' . _t(
-                'SilverStripe\\UserForms\\Model\\UserDefinedForm.FileUploadWarning',
-                'Files uploaded through this field could be publicly accessible if the exact URL is known'
-            )
-            . '</p>'
-        ), 'Type');
+        $folderIsRestricted = Member::actAs(null, function () {
+            return $this->FolderID && !$this->Folder()->canView();
+        });
+
+        if (!$folderIsRestricted) {
+            $fields->addFieldToTab("Root.Main", LiteralField::create(
+                'FileUploadWarning',
+                '<p class="alert alert-warning">' . _t(
+                    'SilverStripe\\UserForms\\Model\\UserDefinedForm.UnrestrictedFileUploadWarning',
+                    'Access to the current upload folder "{path}" is not restricted. Uploaded files will be publicly accessible if the exact URL is known.',
+                    ['path' => $this->Folder()->Filename]
```

Comment:
You're not doing an existence check here. It seems to default back to the root folder somehow, but we probably should have an explicit check.

Reply:
If we don't move the HTML blurb to an SS template file, please escape `$this->Folder()->Filename`. I don't think it's possible to put an XSS attack or a special chracter in a folder name, but better be safe than sorry.